src/usr/hdat/hdatiplparms.C

```diff
@@ -724,18 +725,97 @@ void HdatIplParms::hdatGetSystemParamters()
 
     this->iv_hdatIPLParams->iv_sysParms.hdatSplitCoreMode = 1;
 
-    TARGETING::ATTR_SYSTEM_BRAND_NAME_type l_systemBrandName = {0};
-    if(l_pSysTarget->tryGetAttr<TARGETING::ATTR_SYSTEM_BRAND_NAME>
-                                                         (l_systemBrandName))
+    /* Read UTIL::F5 keyword (maxlen=16)*/
+    errlHndl_t l_err = NULL;
+    size_t  dataSize = 0;
+    char    dataBuffer[17]={0};
+    char    sysVendorname[64]={0};
```

Comment:
I would declare this as 

char sysVendorname[sizeof(this->iv_hdatIPLParams->iv_sysParms.hdatSystemVendorName)]={0};

Or create  symbolic constant for the size, so that that size is defined by exactly one 
constant.  That way if one size changes all the related data types change at the same time.

Reply:
Yes, i want to follow the define of  "this->iv_hdatIPLParams->iv_sysParms.hdatSystemVendorName".
Unfortunately, it also uses a fixed number.
Because I don't want to modify the existing code, I use fixed numbers.
Now I have made changes according to your first suggestion